python/lsst/daf/butler/core/deferredDatasetHandle.py

```diff
@@ -112,3 +116,37 @@ def get(self, parameters: typing.Union[None, dict] = None, **kwargs: dict) -> ty
             mergedParameters = {}
 
         return self._get(mergedParameters)
+
+    def getUri(self, parameters: typing.Union[None, dict] = None, **kwargs: dict) -> typing.Any:
+        """ Retrieves the dataset URI pointed to by this handle
+
+        This handle may be used multiple times, possibly with different
+        parameters.
+
+        Parameters
+        ----------
+        parameters : `dict` or None
+            The parameters argument will be passed to the butler get method.
+            It defaults to None. If the value is not None,  this dict will
+            be merged with the parameters dict used to construct the
+            `DeferredDatasetHandle` class.
+        kwargs : `dict`
+            This argument is deprecated and only exists to support legacy
+            gen2 butler code during migration. It is completely ignored
+            and will be removed in the future.
+
+        Returns
+        -------
+        return : `Object`
+            The dataset pointed to by this handle
+        """
+        if self.parameters is not None:
+            mergedParameters = self.parameters.copy()
+            if parameters is not None:
+                mergedParameters.update(parameters)
+        elif parameters is not None:
+            mergedParameters = parameters
+        else:
+            mergedParameters = {}
+
+        return self._getUri(mergedParameters)
```

Comment:
I don't think it is a good idea to have all the logic for merging parameters duplicated different places, I think if you want to keep this function, that both of these get factored out into a common private method that these both use.

As a side note I have a ticket open to allow fetching one component of a composite dataset given this handle. We should talk about how that may intersect with your ticket.